src/rhash/hash.c

```diff
@@ -679,6 +679,325 @@ static int rc_hash_7800(char hash[33], const uint8_t* buffer, size_t buffer_size
   return rc_hash_buffer(hash, buffer, buffer_size);
 }
 
+struct rc_hash_zip_idx
+{
+  size_t length;
+  uint8_t* data;
+};
+
+static int rc_hash_zip_idx_sort(const void* a, const void* b)
+{
+  struct rc_hash_zip_idx *A = (struct rc_hash_zip_idx*)a, *B = (struct rc_hash_zip_idx*)b;
+  size_t len = (A->length < B->length ? A->length : B->length);
+  return memcmp(A->data, B->data, len);
+}
+
+static int rc_hash_zip_file(md5_state_t* md5, void* file_handle)
+{
+  /* Constants used in this function for reading ZIP and ZIP64 files */
+  const uint8_t  RC_ZIP_END_OF_CENTRAL_DIR_HEADER_SIZE = 22;
+  const uint32_t RC_ZIP_END_OF_CENTRAL_DIR_HEADER_SIG = 0x06054b50;
+  const uint8_t  RC_ZIP_ECDH_CDIR_TOTAL_ENTRIES_OFS = 10;
+  const uint8_t  RC_ZIP_ECDH_CDIR_SIZE_OFS = 12;
+  const uint8_t  RC_ZIP_ECDH_CDIR_OFS_OFS = 16;
+  const uint8_t  RC_ZIP_CENTRAL_DIR_HEADER_SIZE = 46;
+  const uint32_t RC_ZIP_CENTRAL_DIR_HEADER_SIG = 0x02014b50;
+  const uint8_t  RC_ZIP_CDH_METHOD_OFS = 10;
+  const uint8_t  RC_ZIP_CDH_CRC32_OFS = 16;
+  const uint8_t  RC_ZIP_CDH_COMPRESSED_SIZE_OFS = 20;
+  const uint8_t  RC_ZIP_CDH_DECOMPRESSED_SIZE_OFS = 24;
+  const uint8_t  RC_ZIP_CDH_FILENAME_LEN_OFS = 28;
+  const uint8_t  RC_ZIP_CDH_EXTRA_LEN_OFS = 30;
+  const uint8_t  RC_ZIP_CDH_LOCAL_HEADER_OFS = 42;
+  const uint8_t  RC_ZIP_CDH_COMMENT_LEN_OFS = 32;
```

Comment:
Offsets in this file are normally documented with comments instead of constants:

https://github.com/RetroAchievements/rcheevos/blob/3cadf84c30bbc050c0fec79d26e1c8ff504bda42/src/rhash/hash.c#L337-L339

https://github.com/RetroAchievements/rcheevos/blob/3cadf84c30bbc050c0fec79d26e1c8ff504bda42/src/rhash/hash.c#L568-L573

Reply:
In the latest commit I reworked the function a bit and removed the single use constant variables.